aten/src/ATen/native/hammerblade/Where.cpp

```diff
@@ -0,0 +1,34 @@
+#include <ATen/Dispatch.h>
+#include <ATen/hammerblade/HammerBladeContext.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/hammerblade/Offload.h>
+
+namespace at { namespace native {
+
+Tensor _s_where_hb(const Tensor& condition, const Tensor& self, const Tensor& other) {
+    TORCH_CHECK(self.dtype() == other.dtype(), "expected scalar type ", self.dtype(), " but found ", other.dtype());
+    Tensor result = at::empty(self.sizes(), self.options());
+
+    auto iter = at::TensorIterator();
+    iter.set_check_mem_overlap(true);
+    iter.add_output(result);
+    iter.add_input(condition);
+    iter.add_input(self);
+    iter.add_input(other);
+    iter.dont_compute_common_dtype();
+    iter.build();
+
+    if (condition.scalar_type() == at::ScalarType::Byte) {
+        AT_DISPATCH_FLOAT_TYPE_ONLY(iter.dtype(), "where_byte_hb", [&](){
```

Comment:
@zzy82518996 Regarding our discussion of 'AT_DISPATCH_FLOAT_TYPE_ONLY':
I think this works, because it only checks the type of the first tensor (here of output, which is of type float): 
[TensorIterator::dtype()](https://github.com/cornell-brg/hb-pytorch/blob/084cf4d483e4adc2451d6875eb7cafd91d7ea47b/aten/src/ATen/native/TensorIterator.h#L200)

Is this behaviour in general acceptable for you? 

Surely, I can quickly change it to 'AT_DISPATCH_ALL_TYPES' if you would prefer.